applications/crossbar/test/cb_harrys_SUITE.erl

```diff
@@ -0,0 +1,187 @@
+  config().
+end_per_suite(Config) ->
+  ok = inets:stop(),
+  ok = logout_user(),
+  Config.
+
+%% =============================================================================
+%% API tests
+%% =============================================================================
+-spec harrys(Config::config()) ->
+  {comment, string()}.
+harrys(_Config) ->
+  ct:comment("There should be not records yet"),
+  {200, Resp} = api_call(get),
+  <<"success">> = get_value(<<"status">>, Resp),
+  [] = get_value(<<"data">>, Resp),
+
+  ct:comment("Let's create a couple of Harrys"),
+  Harry0 = build_harry(<<"Nonie">>, <<"Dormand">>, <<"ndormand0@dion.ne.jp">>),
+  {201, Resp0} = api_call(put, jiffy:encode(Harry0)),
+  <<"success">> = get_value(<<"status">>, Resp0),
+  <<"Nonie">> = get_value(<<"first_name">>, get_value(<<"data">>, Resp0)),
+
+  Harry1 = build_harry(<<"Kippy">>, <<"Guillain">>, <<"kguillain0@freew.com">>),
+  {201, Resp1} = api_call(put, jiffy:encode(Harry1)),
+  <<"success">> = get_value(<<"status">>, Resp1),
+  Data1 = get_value(<<"data">>, Resp1),
+  <<"kguillain0@freew.com">> = get_value(<<"email">>, Data1),
+  Id1 = get_value(<<"id">>, Data1),
+
+  ct:comment("Let's modify Harry1's email"),
+  NewHarry1 = build_harry(<<"Kippy">>, <<"Guillain">>, <<"kippy@guillain.us">>),
+  {200, NewResp1} = api_call(post, jiffy:encode(NewHarry1), Id1),
+  <<"success">> = get_value(<<"status">>, NewResp1),
+  NewData1 = get_value(<<"data">>, NewResp1),
+  <<"kippy@guillain.us">> = get_value(<<"email">>, NewData1),
+  Id1 = get_value(<<"id">>, NewData1), % match id
+
+  ct:comment("There must be 2 Harrys right now"),
+  {200, Resp2} = api_call(get),
+  2 = get_value(<<"page_size">>, Resp2),
+  2 = length(get_value(<<"data">>, Resp2)),
+
+  ct:comment("Let's delete Harry1"),
+  {200, Resp3} = api_call(delete, [], Id1),
+  Data3 = get_value(<<"data">>, Resp3),
+  true = get_value(<<"deleted">>, get_value(<<"_read_only">>, Data3)),
+  Id1 = get_value(<<"id">>, Data3),
+
+  ct:comment("Trying to delete an unexisting Harry should return 404"),
+  % Harry1 was already deleted on the previous test
+  {404, Resp4} = api_call(delete, [], Id1),
+  <<"error">> = get_value(<<"status">>, Resp4),
+  <<"bad_identifier">> = get_value(<<"message">>, Resp4),
+
+  ct:comment("There must be only 1 Harry left"),
+  {200, Resp5} = api_call(get),
+  1 = get_value(<<"page_size">>, Resp5),
+  1 = length(get_value(<<"data">>, Resp5)),
+
+  {comment, ""}.
+
+%% =============================================================================
+%% Private (helpers)
+%% =============================================================================
+-spec login_user() ->
+  ok.
+login_user() ->
+  URL = "http://192.168.99.100:8000/v2/user_auth",
+  Creds = <<"1141aeae55223480eca2d9e50e3ecb3b">>,
+  ID = <<"harenson">>,
+  % JSON={"data": {"credentials": "value here", "account_name": "value here"}}.
+  ReqBody = jiffy:encode({[{<<"data">>, {[{<<"credentials">>, Creds},
```

Comment:
We don't hard-code the internal representation of JSON. Use kz_json:from_list/1 or kz_json:set_value(s) to create objects.